### DISCUSSION TITLE
Displaying an alert/notification/error to the user

### DISCUSSION THREAD

**Discussion by huyz** [SilentVoid13/Templater#714] [Help] [answered]
Sometimes the template doesn't have everything it needs. In my case, my template needs to work only if correct data is in the clipboard.

How can I prompt the user or display an alert/error/notification (without forcing them to go to the Dev Tools console)?

**Comment by AB1908** [comment, marked as answer]
Use `new Notice("clipboard data missing")` after a clipboard check

**Reply by welpdx** [reply]
Notice() has a second parameter for the duration the toast in miliseconds. 
For example
`new Notice("clipboard data missing",  4000)` has the toast be displayed for 4secs.

You can also pass variables to be displayed like wha you would do with console.log()

**Reply by huyz** [reply]
Where is the documentation for this? I looked and searched and looked…

**Reply by AB1908** [reply]
This isn't templater stuff haha. We're in Obsidian API territory.

**Reply by huyz** [reply]
I guess I forgot to search for `Notice` when I was looking inside https://github.com/obsidianmd/obsidian-api/blob/master/obsidian.d.ts

**Reply by FelipeRearden** [reply]
>Notice() has a second parameter for the duration the toast in miliseconds.
For example
new Notice("clipboard data missing", 4000) has the toast be displayed for 4secs.

>You can also pass variables to be displayed like wha you would do with console.log()

Hello @welpdx !!!!

Just one question about `Notice` ...

**Is it possible to apply a command or URL/URI to be executed when clicking on the Notification ?**

- If YES, is it possible to have different input between `click` and `right click` ?

Thanks for your help!!!!

**Reply by AB1908** [reply]
Yes but what's the use case? For executing commands, there's a private API: `app.commands.executeCommand`.

Also inspect the `commands` attribute. Has some useful stuff.

For differentiating between clicks, we need to capture the `Event` and probably need to add an event listener to the Notice but I'm not sure how to do that.

**Reply by FelipeRearden** [reply]
Hello @AB1908 !!!

Thanks for your reply!

> Yes but what's the use case? 

I use a lot of QuickAdd Macros to automate many workflows on Obsidian. Many times I use this templater script in the last action to know that my MACRO is finished;

```
<%*
new Notice('✓');
%>
```

But not very often I have to add a final command to the same workflow that occurs without a schedule.

My idea is to apply a URI to the Templater script to have the option to use or not the "command" inside. 
- When I say URI I am referring to the plugin Advanced URI.

### **Simple Example:** 

**MACRO 01:** 
a) Send Selection to a specific note;
b) Notification 

**MACRO 02:**
a) Send Selection to a specific note;
b) Notification 
c) Open the destination note in a new pane 

If I could apply a URI to the notification:
- I could transform action **c)** into a URI 
- add it to the action **b)** in MACRO1 
- delete MACRO 2 and execute only  MACRO 1 for 2 different workflows;

I really appreciate if you could help to insert a URI into a templater script that uses `notice` :)

---

> For executing commands, there's a private API: `app.commands.executeCommand`.
> 
> Also inspect the `commands` attribute. Has some useful stuff.
> 

Good to know! We can bring superpowers to  our notifications. 

> For differentiating between clicks, we need to capture the `Event` and probably need to add an event listener to the Notice but I'm not sure how to do that.

Thanks for letting me know, I think I saw this feature into a notification of some plugin, but I am not referring at the moment.

**Reply by AB1908** [reply]
This is a bit much for me to follow so I'll have to get back to this later but happy to chat about ideas over discord. We're mostly templater folks here but people in the Discord might have better ideas. You should also ask on QuickAdd's repo too.

**Reply by FelipeRearden** [reply]
> This is a bit much for me to follow so I'll have to get back to this later but happy to chat about ideas over discord.

Okay, thanks for your reply  and attention to my question! 

> We're mostly templater folks here but people in the Discord might have better ideas. 

I always try to search in Discord for some info too :)

> You should also ask on QuickAdd's repo too.

OK !
